compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -139,23 +139,25 @@ class ExtractDependencies extends Phase {
       ctx.sbtCallback.binaryDependency(file, binaryClassName, fromClassName, sourceFile, context)
 
     def processExternalDependency(depFile: AbstractFile) = {
-      def binaryClassName(classSegments: List[String]) =
-            classSegments.mkString(".").stripSuffix(".class")
+
+      val binaryClassName = ctx.atPhase(ctx.flattenPhase) { implicit ctx =>
```

Comment:
Was the previous mechanism used to find the binary class name incorrect in some situations?